### PR TITLE
ci: test Helm chart against various Kubernetes clusters

### PR DESCRIPTION
**Pull request by ammario**
ci: test Helm chart against various Kubernetes clusters

### CODE REVIEW

File: .github/workflows/k8s.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        dist:
+          - aks
+          - eks
+          - gke
+          # - openshift
+    env:
+        REPLICATED_API_TOKEN: ${{ secrets.REPLICATED_API_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (matifali) [inline .github/workflows/k8s.yaml:55]:
```suggestion
        uses: actions/checkout@v4
```

**Comment by matifali** [comment]
Should we also use the `hcr.io/coder/coder-preview:main` image in these helm charts?
This ould mean we run this action after ci.yaml finishes.

**Comment by ammario** [comment]
@matifali — perhaps later. I'm aiming for MVP right now.

**Comment by ammario** [comment]
**Sep 22 - 01:17PM - 2aa07b6a**

- Have helm install working how I'd like it locally.

File: .github/workflows/k8s.yaml [hunk 2]
```diff
@@ -0,0 +1,137 @@
+# The k8s tests our Kubernetes support by installing Coder via helm against
+# a broad range of Kubernetes distributions.
+#
+# In the future, we should run a subset (or maybe all) of our e2e test suite
+# against each of these clusters.
+name: k8s
+on:
+  push:
+    branches:
+      - main
```
Comment (matifali) [inline .github/workflows/k8s.yaml:10]: Should we use the same paths filter on `main` as used on `pull_request`?

**Comment by ammario** [comment]
Going to close this out as I don't have the time to continue exploring. Maybe someone picks this up as we prioritize e2e tests.

**Pull request closed without merging**